pet-store-vanilla-js/form.js

```diff
@@ -0,0 +1,176 @@
+import { createSubmitSpinner, showError, hideError } from './utils.js';
+import { editPet, addPet } from './api.js';
+import {
+  enablePetModalEvents,
+  disablePetModalEvents,
+  enableModalsBackdropClosing,
+  showDeleteModal,
+  setPetModalHeaderText,
+  disableModalsBackdropClosing,
+  hidePetModal,
+  petModalElements,
+} from './modals.js';
+import { refreshPets } from './app.js';
+
+export const formElements = {
+  form: document.getElementById('pet-modal-form'),
+  petName: document.getElementById('petName'),
+  age: document.getElementById('age'),
+  notes: document.getElementById('notes'),
+  kind: document.getElementById('kind'),
+  healthProblems: document.getElementById('healthProblems'),
+  addedDate: document.getElementById('addedDate'),
+
+  saveButton: document.getElementById('form-save-btn'),
+  editButton: document.getElementById('form-edit-btn'),
+  deleteButton: document.getElementById('form-delete-btn'),
+  cancelButton: document.getElementById('form-cancel-btn'),
+  lockButton: document.getElementById('form-lock-btn'),
+};
+
+formElements.form.addEventListener(
+  'submit',
+  async function handleFormSubmit(e) {
+    e.preventDefault();
+
+    disablePetModalEvents();
+    showFormSubmitSpinner();
+
+    const pet = getFormValues(e.target);
+    try {
+      let petResponse;
+      if (Number(pet.petId) > 0) {
+        petResponse = await editPet(pet);
+      } else {
+        petResponse = await addPet(pet);
+      }
+
+      await refreshPets();
```

Comment:
You shouldn't be waiting for the pets to re-load before finishing this async operation (hiding the spinner & locking the form).